### PR TITLE
support param sync validation for QWen-MoE.

### PR DESCRIPTION
**Pull request by charles9304**

### CODE REVIEW

File: chatlearn/models/base_module.py [hunk 1]
```diff
@@ -728,7 +730,8 @@ def allgather_expert_parameter(self, group_name, pipe_stage=0):
         for name, param in self._parameters_to_sync[pipe_stage]:
             param, state = self._synchronizer.regroup_experts_from_all_tp_ranks(name, param, self.tensor_parallel_rank(), group_name)
             if state:
-                self._expert_sync_buffer[name].append(param)
+                self._expert_sync_buffer.pop(name, "Not Found.")
```
Comment (SeaOfOcean) [inline chatlearn/models/base_module.py:733]: should we throw an exception if name is not found

Comment (adoda) [inline chatlearn/models/base_module.py:733]: 这里是什么用途？

Comment (charles9304) [reply chatlearn/models/base_module.py:733]: `pop` here aims to release some weight in _expert_sync_buffer, if not found, pass is ok

Comment (charles9304) [reply chatlearn/models/base_module.py:733]: `pop`  here aims to release memory of some weight in _expert_sync_buffer, if not found, pass is ok.

**Review by SeaOfOcean — APPROVED** [review]

**Review by adoda — APPROVED** [review]

**Pull request merged**